### DISCUSSION TITLE
Expose an "applied_revision" prometheus metric

### DISCUSSION THREAD

**Discussion by romainmeunier** [fluxcd/flux2#1796] [Proposals] [answered]
We currently have this reconciliation metric  : 

`gotk_reconcile_condition{kind="Kustomization",name="flux-system",namespace="flux-system",status="True",type="Ready"} 1`

This metric doesn't expose the revision, which is a crucial information, especially in a multi-tenancy environment.

**Comment by seh** [comment]
That's likely out of concern for the high cardinality of such a metric label's values.

**Comment by kingdonb** [comment, marked as answer]
Please refer to:

* https://github.com/fluxcd/flux2/issues/4128

This has been addressed, now you can create any metrics that you wish if the information is in the CRD spec or status, you can create a metric to report on it.

See also:

https://fluxcd.io/flux/monitoring/custom-metrics/
